Additional google analytics events

NoahCarnahan opened the pull request:
Add google analytics events for the following:
- Change case list/detail property format
- Download Form Export
- Download Case Export

(The commit list here is confusing, sorry bout that)

NoahCarnahan committed:
Add case property format change event

NoahCarnahan committed:
Add case property format change event

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into moar-analyics-events

NoahCarnahan committed:
Track form and case export downloads

NoahCarnahan committed:
Add doc string

NoahCarnahan committed:
Rename some parameters

NoahCarnahan committed:
Pre merge commit: Refactored export manager slightly to make merge easier

NoahCarnahan committed:
Merge branch 'moar-analytics-events' into merge-test-halp

Conflicts:
	corehq/apps/reports/static/reports/ko/export.manager.js

NoahCarnahan committed:
Fix note

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -35,12 +35,24 @@ var ExportManager = function (o) {
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Use '===' to compare with '0'.

NoahCarnahan commented:
Waiting for a confirmation on the spec

NoahCarnahan commented on corehq/apps/reports/static/reports/javascripts/reports.download_export.js:
```diff
@@ -36,4 +37,14 @@ var HQExportDownloader = function (options) {
+     * Attach a click event handler to the download modal "download" button that
+     * notifies google of the download event.
+     */
+    self.setUpEventTracking = function(){
```
You might notice that this function is very similar to the identically named `setUpEventTracking` function on `ExportManager` object. This object (`HQExportDownloader`) duplicates some functionality of `ExportManager` and exists on the same page. `HQExportDownloader` should probably be done away with. Seems beyond the scope of this feature though.

NoahCarnahan commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -54,14 +66,6 @@ var ExportManager = function (o) {
-                    "Download Mobile Device Log",
-                    "Export Mobile Device Log"
-                );
-            }
```
This block has been moved into the `setUpEventTracking` function.

NoahCarnahan commented on corehq/apps/reports/templates/reports/reportdata/case_export_data.html:
```diff
@@ -79,7 +80,11 @@
+                           href="#export-download-status"
+                           data-backdrop="static"
+                           data-formname="{{ export.name }}"
+                           data-dlocation='{% url "export_custom_data" domain export.get_id %}?type=case&{{ get_filter_params.urlencode|safe }}&async=true'
```
(just making this one more readable)

nickpell commented:
looks good to merge when spec is confirmed.

NoahCarnahan committed:
Alter the types of GA events that various downloads trigger

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -35,12 +36,24 @@ var ExportManager = function (o) {
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Use '===' to compare with '0'.

dimagimon replied:
Use '===' to compare with '0'.

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -83,6 +88,50 @@ var ExportManager = function (o) {
+            var category, action;
+            if (self.export_type == "form"){
+                category = "Form Exports";
+                action = "Download (any) Form Export"
```
Missing semicolon.

NoahCarnahan committed:
fix missing semicolon

NoahCarnahan committed:
Reworked "Download Custom Case Export" event tracking

NoahCarnahan committed:
Reworked "Download Saved Case Export" event tracking

NoahCarnahan committed:
Rework "Download Form Export" events

- The raw download event uses the form name as the label
- The custom, saved, and deidentified downloads use the form xmlns
  as the event label.

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -35,12 +36,25 @@ var ExportManager = function (o) {
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Use '===' to compare with '0'.

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -83,6 +89,44 @@ var ExportManager = function (o) {
+
+
+                var action = "Download Raw Form Export";
+                var label = params.isBulkDownload ? "bulk" : params.formName;
```
'label' is already defined.

NoahCarnahan committed:
clean lint

NoahCarnahan committed:
Use xmlns for raw form download event label

NoahCarnahan committed:
Rework form export events again :)

dimagimon commented on corehq/apps/reports/static/reports/ko/export.manager.js:
```diff
@@ -35,12 +36,26 @@ var ExportManager = function (o) {
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Use '===' to compare with '0'.

NoahCarnahan commented:
@nickpell spec has finally settled :) This should be good to go.
